src/sql/workbench/parts/objectExplorer/browser/iconRenderer.ts

```diff
@@ -0,0 +1,304 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { generateUuid } from 'vs/base/common/uuid';
+
+export function renderServerIcon(element: HTMLElement, iconPath: IconPath, isConnected: boolean,
+	iconSizePx?: number, topMarginPx?: number, leftMarginPx?: number
+): void {
+	if (!element) { return; }
+	renderIcon(element, iconPath);
+	let badge: ServerStatusBadge = new ServerStatusBadge(iconSizePx, topMarginPx, leftMarginPx);
+	badge.renderBadge(element, isConnected);
+}
+
+export function renderIcon(element: HTMLElement, iconPath: string | IconPath): void {
+	if (!element || !iconPath) { return; }
+
+	let lightPath: string = undefined;
+	let darkPath: string = undefined;
+	if (typeof iconPath === 'string') {
+		lightPath = iconPath;
+		darkPath = iconPath;
+	} else {
+		lightPath = iconPath.light;
+		darkPath = iconPath.dark;
+	}
+
+	if (element.id === undefined || element.id === '') {
+		element.id = `id_${generateUuid()}`;
+	}
+	let elementId: string = element.id;
+	let styleId: string = `icon_style_${elementId}`;
+
+	let current = element.previousElementSibling;
+	while (current) {
+		let next = current.previousElementSibling;
+		if (current.id === styleId) {
+			current.remove();
+			break;
+		}
+		current = next;
+	}
+
+	element.insertAdjacentHTML('beforebegin',
+		`<style type="text/css" id="${styleId}">
+			.monaco-shell #${elementId} {
+				background: url('${lightPath}') center center no-repeat;
+			}
+			.vs-dark #${elementId},
+			.hc-black #${elementId} {
+				background: url('${darkPath}') center center no-repeat;
+			}
+		</style>`.replace(/\t/g, ' ').replace(/\r?\n/g, ' ').replace(/  +/g, ' ')
+	);
+}
+
+const serverIconDir: string = `${__dirname}/../../../../media/icons`;
```

Comment:
So this is really frail - we've refactored the location of these files a bunch, and doing this kind of relative import midway in a file has me worried. What if we move the file again (likely to happen)?
- Please fallback to existing behavior, where in `renderConnection` we assign a class and the css in src/sql/workbench/parts/connection/browser/media/connectionViewlet.css decides what to show
  - Update that css so it uses your new default icons instead. 
  - I think you need to move your default icons under src/sql/workbench/parts/connection/browser/media and remove the old connected/disconnected icons from there
  - I think your code already handles badging, so given this we should be fine once this is CSS handled instead of relying on the relative path?

Reply:
removed.